polars/polars-lazy/polars-plan/src/dsl/dt.rs

```diff
@@ -180,6 +180,40 @@ impl DateLikeNameSpace {
             .map_private(FunctionExpr::TemporalExpr(TemporalFunction::TimeStamp(tu)))
     }
 
+    pub fn truncate(self, every: &str, offset: &str) -> Expr {
+        let every = Duration::parse(every);
+        let offset = Duration::parse(offset);
+        self.0.apply(
+            move |s| match s.dtype() {
+                DataType::Datetime(_, _) => {
+                    Ok(s.datetime().unwrap().truncate(every, offset).into_series())
+                }
+                DataType::Date => Ok(s.date().unwrap().truncate(every, offset).into_series()),
+                dt => Err(PolarsError::ComputeError(
+                    format!("expected date/datetime got {:?}", dt).into(),
+                )),
+            },
+            GetOutput::same_type(),
+        )
+    }
+
+    pub fn round(self, every: &str, offset: &str) -> Expr {
+        let every = Duration::parse(every);
+        let offset = Duration::parse(offset);
+        self.0.apply(
```

Comment:
This should be a `map` and we should make a `map_private`. The actual function implementation should be done on the enum dispatch, not inline.

Reply:
Ah sure thing understood. Is this ok?